corehq/ex-submodules/casexml/apps/case/models.py

```diff
@@ -614,10 +614,8 @@ def _apply_action(self, action, xform):
             self.apply_close(action)
         elif action.action_type == const.CASE_ACTION_ATTACHMENT:
             self.apply_attachments(action, xform)
-        elif action.action_type == const.CASE_ACTION_COMMTRACK:
-            pass  # no action needed here, it's just a placeholder stub
-        elif action.action_type == const.CASE_ACTION_REBUILD:
-            pass
+        elif action.action_type in (const.CASE_ACTION_COMMTRACK, const.CASE_ACTION_REBUILD):
+            return  # no action needed here, it's just a placeholder stub
```

Comment:
the issue here was that `user_id` and `modified_on` were being set below even if the action was a stub action that shouldn't set them.